Frends.SFTP.ListFiles/Frends.SFTP.ListFiles/Frends.SFTP.ListFiles.cs

```diff
@@ -0,0 +1,105 @@
+                {
+                    client.Connect();
+                    result = GetFiles(client, regexStr, options.Directory, options, cancellationToken);
+                    client.Disconnect();
+
+                }
+            }
+            catch (SshConnectionException ex)
+            {
+                throw new Exception($"Error when establishing connection to the Server: {ex.Message}", ex);
+            }
+            catch (SocketException ex)
+            {
+                throw new Exception($"Unable to establish the socket: No such host is known.", ex);
+            }
+            catch (SshAuthenticationException ex)
+            {
+                throw new Exception($"Authentication of SSH session failed: {ex.Message}", ex);
+            }
+
+            return result;
+        }
+
+        private static List<Result> GetFiles(SftpClient sftp, string regexStr, string directory, Options options, CancellationToken cancellationToken)
+        {
+            var directoryList = new List<Result>();
+
+            var files = sftp.ListDirectory(directory);
+
+            foreach (var file in files)
+            {
+                if (file.Name != "." && file.Name != "..")
+                {
+                    cancellationToken.ThrowIfCancellationRequested();
+
+                    if (options.IncludeType == IncludeType.Both
+                        || (file.IsDirectory && options.IncludeType == IncludeType.Directory)
+                        || (file.IsRegularFile && options.IncludeType == IncludeType.File))
+                    {
+                        if (Regex.IsMatch(file.Name, regexStr, RegexOptions.IgnoreCase))
+                            directoryList.Add(new Result(file));
+                    }
+
+                    if (file.IsDirectory && options.IncludeSubdirectories)
+                    {
+                        directoryList.AddRange(GetFiles(sftp, regexStr, file.FullName, options, cancellationToken));
+                    }
+                }
+            }
+            return directoryList;
+        }
+
+        // Helper method to create connection info
+        private static ConnectionInfo GetConnectionInfo(Connection connect)
+        {
+            switch (connect.Authentication)
+            {
+                case AuthenticationType.PrivateKey:
+                    return new PrivateKeyConnectionInfo(connect.Address, connect.Port, connect.Username, new PrivateKeyFile(connect.PrivateKeyFileName));
+
+                case AuthenticationType.PrivateKeyPassphrase:
+                    return new PrivateKeyConnectionInfo(connect.Address, connect.Port, connect.Username, new PrivateKeyFile(connect.PrivateKeyFileName, connect.Passphrase));
+
+                default:
+                    return new PasswordConnectionInfo(connect.Address, connect.Port, connect.Username, connect.Password);
+            }
+        }
+
+        private static string WildCardToRegex(string value)
```

Comment:
I don't think it is necessary to create this helper method, since it is only used once in the code and it only executes one line of code.

Reply:
Fixed